nucliadb/src/nucliadb/search/api/v1/hydrate.py

```diff
@@ -0,0 +1,758 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+import base64
+from typing import Literal, Optional, Union, cast, overload
+
+from fastapi import Request, Response
+from fastapi_versioning import version
+
+from nucliadb.common.ids import FIELD_TYPE_STR_TO_NAME, FieldId, ParagraphId
+from nucliadb.common.maindb.utils import get_driver
+from nucliadb.common.models_utils import from_proto
+from nucliadb.ingest.fields.base import Field
+from nucliadb.ingest.fields.file import File
+from nucliadb.ingest.orm.resource import Resource
+from nucliadb.ingest.serialize import get_orm_resource
+from nucliadb.search import SERVICE_NAME
+from nucliadb.search.api.v1.router import KB_PREFIX, api
+from nucliadb.search.search import paragraphs
+from nucliadb.search.search.cache import request_caches
+from nucliadb.search.search.hydrator import hydrate_field_text
+from nucliadb_models import hydration as hydration_models
+from nucliadb_models.common import FieldTypeName
+from nucliadb_models.hydration import Hydrated, HydrateRequest, Hydration
+from nucliadb_models.resource import NucliaDBRoles
+from nucliadb_models.search import Image
+from nucliadb_models.security import ResourceSecurity
+from nucliadb_protos import resources_pb2
+from nucliadb_protos.resources_pb2 import FieldComputedMetadata
+from nucliadb_utils.authentication import requires
+from nucliadb_utils.utilities import get_storage
+
+
+@api.post(
+    f"/{KB_PREFIX}/{{kbid}}/hydrate",
```

Comment:
For this to work in the cluster, you'll need to add these to the virtual service.

I wonder if we should have the endpoint in the reader component instead, as it is purely fetching stuff from maindb and storage. wdyt?

Reply:
I was more concerned in terms of memory usage. but we can work on this once it is a problem